gprofiler/profilers/java.py

```diff
@@ -236,11 +236,16 @@ def __init__(self, stop_event: Event, jattach_timeout: int):
         self.jattach_timeout = jattach_timeout
 
     def run(self, process: Process, cmd: str) -> str:
-        return run_process(
-            [asprof_path(), "jcmd", "--jattach-cmd", cmd, str(process.pid)],
-            stop_event=self.stop_event,
-            timeout=self.jattach_timeout,
-        ).stdout.decode()
+        try:
+            return run_process(
+                [asprof_path(), "jcmd", "--jattach-cmd", cmd, str(process.pid)],
+                stop_event=self.stop_event,
+                timeout=self.jattach_timeout,
+            ).stdout.decode()
+        except CalledProcessError as e:
+            if f"Process {process.pid} not found" in str(e):
```

Comment:
I'm not sure this works. The string `Process xx not found` appears in stderr. Is that present in `str(e)`? Perhaps you need `in e.stderr`?

Reply:
This should work. I checked it and it is present in str(e):
```
>>> from gprofiler.utils import run_process
>>> from gprofiler.exceptions import CalledProcessError
>>> cmd = ["python", "/home/mpozniak/test.py"]
>>> try:
...   run_process(cmd)
... except CalledProcessError as e:
...   print(str(e))
...
Command ['python', '/home/mpozniak/test.py'] returned non-zero exit status 1.
stdout:
stderr: ERROR!
```

and test.py was just sys.exit("ERROR!")